pkg/virt-controller/watch/network.go

```diff
@@ -91,19 +93,37 @@ func trimDoneInterfaceRequests(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInst
 	vm.Status.InterfaceRequests = updateIfaceRequests
 }
 
-func handleDynamicInterfaceRequests(vm *v1.VirtualMachine) {
+func handleDynamicInterfaceRequests(vm *v1.VirtualMachine, vmiHasPodNetwork bool, clusterConfig *virtconfig.ClusterConfig) error {
 	if len(vm.Status.InterfaceRequests) == 0 {
-		return
+		return nil
 	}
 
 	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
+
+	// Explicitly add the default interface before hot-plugging another interface
+	if vmiHasPodNetwork && len(vmTemplateCopy.Domain.Devices.Interfaces) == 0 {
+		var hasAddRequest bool
+		for _, req := range vm.Status.InterfaceRequests {
+			if req.AddInterfaceOptions != nil {
+				hasAddRequest = true
+				break
+			}
+		}
```

Comment:
Extracting this to a `hasAddRequest` function helper would have made this much nicer, avoiding that variable and break.

You seem to have done something similar with `vmiHasPodNetwork` already.

Reply:
I agree, extracting this block to function would make it easier to follow.